dynamodb.tf

```diff
@@ -0,0 +1,26 @@
+resource "aws_dynamodb_table" "geff_batch_locking_table" {
+  count        = var.create_dynamodb_table ? 1 : 0
+  name         = var.user_managed_dynamodb_table_name != null ? var.user_managed_dynamodb_table_name : "${local.geff_prefix}_batch_locking_table" 
+  billing_mode = "PAY_PER_REQUEST"
+  hash_key     = "batch_id"
+
+  attribute {
+    name = "batch_id"
+    type = "S"
+  }
+
+  ttl {
+    attribute_name = "ttl"
+    enabled        = true
+  }
+}
+
+data "aws_dynamodb_table" "user_managed_table" {
+  count = var.user_managed_dynamodb_table_name != null && !var.create_dynamodb_table ? 1 : 0
```

Comment:
does the first part do anything now? afaict, the predicate could be `!var.create_dynamodb_table`?

Reply:
I think we'd have to check for both of these, for a case where the user sets `create_dynamodb_table` to be false, and `batch_locking_table_name` to be null.